src/http/routes.cc

```diff
@@ -115,9 +112,7 @@ future<std::unique_ptr<http::reply> > routes::handle(const sstring& path, std::u
             auto r =  handler->handle(path, std::move(req), std::move(rep));
             return r.handle_exception(_general_handler);
         } catch (const redirect_exception& _e) {
-            rep.reset(new http::reply());
-            rep->add_header("Location", _e.url).set_status(_e.status()).done(
-                    "json");
+            rep.reset(new http::reply(_e.to_reply()));
```

Comment:
nit: 
```suggestion
            rep = std::make_unique<http::reply>(_e.to_reply());
```
for simmetry with line 50

or, if it compiles,
```suggestion
            *rep = _e.to_reply();
```

Reply:
Changed.